python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -52,6 +52,10 @@ def lazyLoadJobApplication(this_job):
 
 def lazyLoadJobObject(this_job, this_attr):
 
+    if stripProxy(this_job)._getRegistry():
+        if stripProxy(this_job)._getRegistry().has_loaded(stripProxy(this_job)):
+            return getattr(this_job, this_attr)
```

Comment:
Not intrinsically bad, no, but a layering violation of sorts. Keeping a clear separation of proxy and non-proxy is a good idea. Maybe in this case it would be served better by a simple `job = stripProxy(this_job)` on the first line of the function to reduce the number of overall calls?